File: src/NomsCiutatsJava/Fase4.java
/**
 * Fase4 - Ejercico de Noms Ciutats
 * @author Rubén Rodríguez
 * IT Academy
 * Barcelona Activa
 *
 */
package NomsCiutatsJava;

import java.util.Arrays;

public class Fase4 {
	
	private char[] ciutat1;
	private char[] ciutat2;
	private char[] ciutat3;
	private char[] ciutat4;
	private char[] ciutat5;
	private char[] ciutat6;
	
	/*
	 * Método que inicializa los arrays de caacteres a la longitud de cada población 
	 * introducida. 
	 */
	private void nousArray(Fase3 fase3) {
		
		byte i=0;
		ciutat1 = new char[(fase3.getArrayCiutat()[0].length())];
		ciutat2 = new char[(fase3.getArrayCiutat()[1].length())];
		ciutat3 = new char[(fase3.getArrayCiutat()[2].length())];
		ciutat4 = new char[(fase3.getArrayCiutat()[3].length())];
		ciutat5 = new char[(fase3.getArrayCiutat()[4].length())];
		ciutat6 = new char[(fase3.getArrayCiutat()[5].length())];

		//Letra a letra agregamos el nombre de la ciudad (string) al array de char.
		
		for (char caracter: ciutat1 ) {
			ciutat1[i]= fase3.getArrayCiutat()[0].charAt(i);
			i++;
		}
		
		i=0;
		for (char caracter: ciutat2 ) {
			ciutat2[i]= fase3.getArrayCiutat()[1].charAt(i);
			i++;
		}
		
		i=0;
		for (char caracter: ciutat3 ) { 
			ciutat3[i]= fase3.getArrayCiutat()[2].charAt(i);
			i++;
		}
		
		i=0;
		for (char caracter: ciutat4 ) { 
			ciutat4[i]= fase3.getArrayCiutat()[3].charAt(i);
			i++;
		}	
		i=0;
		for (char caracter: ciutat5 ) {
			ciutat5[i]= fase3.getArrayCiutat()[4].charAt(i);
			i++;
		}
		
		i=0;
		
		for (char caracter: ciutat6 ) { 
			ciutat6[i]= fase3.getArrayCiutat()[5].charAt(i);
		i++;
		}
	}
	
	
	protected void fase4Print(Fase3 fase3) {

		nousArray(fase3);
		
		printArrayInvertido(ciutat1);
		printArrayInvertido(ciutat2);
		printArrayInvertido(ciutat3);
		printArrayInvertido(ciutat4);
		printArrayInvertido(ciutat5);
		printArrayInvertido(ciutat6);

		
	
	}
	
	/*
	 * Método que invierte un array de char por medio de una iteración 'for'
	 */
	public static void printArrayInvertido(char[] array ) {
		for (int i = array.length-1; i >= 0 ; i--){
		    System.out.print(array[i]);
		}
		System.out.println();
		
    }
}

File: src/NomsCiutatsJavaScript/Fase4.js
/**
 * Fase4 - Ejercico de Noms Ciutats en JavaScript
 * @author Rubén Rodríguez
 * IT Academy
 * Barcelona Activa
 *
 */

let impresF4 = false


var respostaFase4 = function(){
	let selectElement = document.getElementById('fase4Sortida')

	let ciutat_1 = []
	let ciutat_2 = []
	let ciutat_3 = []
	let ciutat_4 = []
	let ciutat_5 = []
	let ciutat_6 = []


	
	selectElement.innerHTML += "<br>Listado del Array Fase4: " 

	if (impresF4) {}
	else {

		impresF4 = true
		let ciutat=""
			
		//let formCiutat = document.formulario.ciutat1.value.split('')
			
		/*
		 * Recorremos uno a uno los caracteres del string almacenado en los campos 
		 * de nombre de ciudad. Y le asignamos al array con el métod push.
		 */
		for (let i = 0; i < document.formulario.ciutat1.value.length; i++) {
			  ciutat_1.push((document.formulario.ciutat1.value.charAt(i)));

		}
		
		for (let i = 0; i < document.formulario.ciutat2.value.length; i++) {
			  ciutat_2.push((document.formulario.ciutat2.value.charAt(i)));

		}

		
		for (let i = 0; i < document.formulario.ciutat3.value.length; i++) {
			  ciutat_3.push((document.formulario.ciutat3.value.charAt(i)));

		}
		
		for (let i = 0; i < document.formulario.ciutat4.value.length; i++) {
			  ciutat_4.push((document.formulario.ciutat4.value.charAt(i)));

		}
			
		for (let i = 0; i < document.formulario.ciutat5.value.length; i++) {
			  ciutat_5.push((document.formulario.ciutat5.value.charAt(i)));

		}
		for (let i = 0; i < document.formulario.ciutat6.value.length; i++) {
			  ciutat_6.push((document.formulario.ciutat6.value.charAt(i)));

		}

		/*
		 * el método reverse nos permite invertir el array
		 * 
		 */
		selectElement.innerHTML += "<BR> "+ ciutat_1.reverse()
		selectElement.innerHTML += "<BR> "+ ciutat_2.reverse()
		selectElement.innerHTML += "<BR> "+ ciutat_3.reverse()
		selectElement.innerHTML += "<BR> "+ ciutat_4.reverse()
		selectElement.innerHTML += "<BR> "+ ciutat_5.reverse()
		selectElement.innerHTML += "<BR> "+ ciutat_6.reverse()


	
		document.getElementById('4').disabled = true

	
	}
}

//Nos aseguramos de cargar todo el JS al cargar el document html.


if (document.addEventListener){
    window.addEventListener('load',(function() {
    	document.getElementById('4').onclick = respostaFase4;
    }),false);
} else {
    window.attachEvent('onload',(function() {
    	document.getElementById('4').onclick = respostaFase4;
    }));    
}